src/main/java/com/minecolonies/coremod/entity/ai/citizen/cook/EntityAIWorkCook.java

```diff
@@ -327,15 +334,14 @@ else if (!InventoryUtils.hasItemInProvider(getOwnBuilding(), stack -> ItemStackU
      */
     private boolean isItemStackForAssistant(ItemStack stack)
     {
-        if(this.assistantTests.isEmpty())
+        if(this.reservedItemCache.isEmpty())
         {
-            this.assistantTests.addAll(getOwnBuilding().getFirstModuleOccurance(BuildingCook.CraftingModule.class).getAssistantItems());
+            this.reservedItemCache.addAll(getOwnBuilding().getFirstModuleOccurance(BuildingCook.CraftingModule.class).reservedStacks().keySet());
         }
 
-        return ItemStackUtils.compareItemStackListIgnoreStackSize(new ArrayList<>(assistantTests), stack, true, true);
+        return reservedItemCache.contains(new ItemStorage(stack));
```

Comment:
Unless `reservedStacks()` returns storages with the ignore NBT flag set, this will no longer ignore NBT.  For most crafting ingredients that may be a good thing, but I suspect it might break compatibility with utensils that take damage (unless those already weren't supported due to other code).

Reply:
This is only for "cook not taking away cooking materials", so this should not be a problem with taking damage ones?